pkg/patch/patch_test.go

```diff
@@ -73,14 +74,75 @@ func TestGeneratePatch_AppendSidecarTrue(t *testing.T) {
 		t.Fatal("invalid json")
 	}
 
+	verifyInitContainerPatch(t, string(patch))
 	verifyPatch(t, string(patch), meta)
 }
 
-func verifyPatch(t *testing.T, patch string, meta Meta) {
+func TestGeneratePatch_AppendSidecarTrue_WithAppMeshCNI(t *testing.T) {
+	meta := Meta{
+		AppendImagePullSecret: false,
+		HasImagePullSecret:    false,
+		AppendSidecar:         true,
+		AppendInit:            false,
+		Init: InitMeta{
+			Ports:              "80,443",
+			EgressIgnoredPorts: "22",
+			ContainerImage:     "111345817488.dkr.ecr.us-west-2.amazonaws.com/aws-appmesh-proxy-route-manager:v2",
+			IgnoredIPs:         "169.254.169.254",
+		},
+		Sidecar: SidecarMeta{
+			MeshName:          "global",
+			VirtualNodeName:   "podinfo",
+			Region:            "us-west-2",
+			LogLevel:          "debug",
+			ContainerImage:    "111345817488.dkr.ecr.us-west-2.amazonaws.com/aws-appmesh-envoy:latest",
+			CpuRequests:       "10m",
+			MemoryRequests:    "32Mi",
+			InjectXraySidecar: true,
+			EnableStatsTags:   true,
+		},
+		PodAnnotations: map[string]string{
+			"appmesh.k8s.aws/appmeshCNI": "enabled",
+		},
+	}
+
+	patch, err := GeneratePatch(meta)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !json.Valid([]byte(patch)) {
+		t.Fatal("invalid json")
+	}
+
+	verifyPatch(t, string(patch), meta)
+}
+
+func verifyInitContainerPatch(t *testing.T, patch string) {
 	if !strings.Contains(patch, "aws-appmesh-proxy-route-manager:v2") {
 		t.Errorf("Init container image not found")
 	}
+}
+
+func verifyAppMeshCNIAnnotationsPatch(t *testing.T, patch string) {
```

Comment:
I don't see where this function is being called.

Reply:
oops! I was refactoring the tests and missed adding this check for the new test I wrote. Fixed it.